src/bbd/geocoder/geocoder.py

```diff
@@ -0,0 +1,582 @@
+            str and setting defaults as necessary. (see defaults)
+        For dataframe: 
+            data should either contain 1 column
+            OR 
+            columns using follow VAN default column names (not case sensitive):
+             "Address" - column of str street component of address
+             "City" - column of str city component of address
+             "County" - column of str county component of address
+             "State" - column of str state component of address
+             "Zip5" - column of str 5-digit postal code for address
+             "Country" - column of str country component of address
+                         (if applicable, default will set country to US)
+        For list of dicts:
+            Each item should be a dict of the form
+             "Address" : str street component of address
+             "City" : str city component of address
+             "County" : str county component of address
+             "State" : str state component of address
+             "Zip5" : str 5-digit postal code for address
+             "Country" : str country component address
+                         (if applicable, default will set country to US)
+        For list of strings:
+            Each item should str full address to pass as query.
+
+    email : str
+        A valid email address as str to pass to Nominatim as user_agent.
+        See Nominatim's Usage Policy.
+
+    path : str or pathlib Path object
+        A path to where geocoded data will be saved.
+
+    batch_size : (Optional) int
+        Number of batches to geocode at a time. Useful for running subset of
+        data. Will determin size of batch for run(num_batches) and size of 
+        progress bar while running.
+
+    defaults : (Optional) dict
+        A dictionary of default values for query. Should be of form:
+        {
+            "Address" : address_default,
+            "City" : city_default,
+            "County" : county_default,
+            "State" : state_default,
+            "Zip5" : postal_default,
+            "Country" : country_default
+        }
+        Will set country to United States by default.
+
+    keep_index : (Optional) bool
+        Whether to use the index of data for the geocoded result. If 
+        False, will use 
+        It is recommended to use the same index for the inputted data
+        to easily merge the data.
+        True by default.
+
+    Properties
+    ----------
+    street_encode : func
+        A function of 1 argument used to encode the street address. The 
+        default is street_encode(address) used to remove secondary 
+        descriptors from the street address. This is recommended 
+        because many addresses in OpenStreetMaps do not have
+        secondary descriptors attached and therefore will not be 
+        geocoded.
+
+    geocoder : str
+        Enter email as str to change geocoder user_agent.
+
+    email : str
+        Alias for geocoder
```

Comment:
email as alias for geocoder stemmed from the aforementioned going ham on properties and having an more intuitive representation of it. So changing email is effectively setting the new geocoder (an instance of `geopy.geocoders.Nominatim` with user_agent = email wrapped in a `geopy.extras.RateLimiter`)

Reply:
Yeah I'm updating it in the new branch. email will still be an alias but the docstring description will be more accurate. I still want it to be an alias b/c I want changing the email to change the geocoder (otherwise changing email does nothing). Idk maybe I will just separate it into 2 properties the only problem is either way changing email will change geocoder.